core/src/main/java/org/apache/calcite/util/graph/Graphs.java

```diff
@@ -56,42 +54,40 @@ public int size() {
   public static <V, E extends DefaultEdge> FrozenGraph<V, E> makeImmutable(
       DirectedGraph<V, E> graph) {
     DefaultDirectedGraph<V, E> graph1 = (DefaultDirectedGraph<V, E>) graph;
-    Map<Pair<V, V>, List<V>> shortestPaths = new HashMap<>();
+    Map<Pair<V, V>, int[]> shortestDistances = new HashMap<>();
     for (DefaultDirectedGraph.VertexInfo<V, E> arc
         : graph1.vertexMap.values()) {
       for (E edge : arc.outEdges) {
         final V source = graph1.source(edge);
         final V target = graph1.target(edge);
-        shortestPaths.put(Pair.of(source, target),
-            ImmutableList.of(source, target));
+        shortestDistances.put(Pair.of(source, target), new int[] {1});
       }
     }
     while (true) {
       // Take a copy of the map's keys to avoid
       // ConcurrentModificationExceptions.
       final List<Pair<V, V>> previous =
-          ImmutableList.copyOf(shortestPaths.keySet());
-      int changeCount = 0;
+          ImmutableList.copyOf(shortestDistances.keySet());
+      boolean changeCount = false;
```

Comment:
minor thing: now that this variable is a `boolean`, maybe it should be renamed (`changeCount` -> `change` ?)

Reply:
Revised. Thanks for your kind reminder.